homeassistant/components/shelly/event.py

```diff
@@ -181,13 +218,49 @@ def __init__(
     async def async_added_to_hass(self) -> None:
         """When entity is added to hass."""
         await super().async_added_to_hass()
+
         self.async_on_remove(
             self.coordinator.async_subscribe_input_events(self._async_handle_event)
         )
 
     @callback
     def _async_handle_event(self, event: dict[str, Any]) -> None:
         """Handle the demo button event."""
-        if event["id"] == self.input_index:
+        if event["id"] == self.event_id:
             self._trigger_event(event["event"])
```

Comment:
```suggestion
            self._trigger_event(event[ATTR_EVENT])
```

Reply:
I don't think we should use `ATTR_EVENT`, this is something internal to Shelly, not core attribute